client/consensus/babe/src/authorship.rs

```diff
@@ -32,6 +32,9 @@ use sp_core::Pair;
 use sc_keystore::KeyStorePtr;
 use super::Epoch;
 
+/// BABE VRFInOut context.
+pub(super) static BABE_VRF_INOUT_CONTEXT: &[u8] = b"BabeVRFInOutContext";
```

Comment:
Would it make sense to just re-use the same context we are using when checking primary threshold?

Reply:
Meh.  If you do so, then the threshold check on https://github.com/paritytech/substrate/blob/48b124008a6afc296cf310c118164ac85b13e515/client/consensus/babe/src/authorship.rs#L98 would reduces the entropy here by log2(threshold).  We've quite small thresholds, so that's okay, but why?   

If you invoke `make_bytes` twice then you might as well use different contexts.  If you combine the `make_bytes` invocations then sure, but also we could add some `VRFInOut::fill_bytes` method since `make_bytes::<[u8;40]>()` would need stable const generics.  

These are roughly shake256 invocations that should be quite fast, well maybe the hash-to-curve in creating the `VRFInOut` takes longer, but still fast compared to even checking a signature.

Reply:
@burdges why are we using shake256 instead of something like blake2b + chacha20?

Reply:
`merlin::Transcript` gives a cleaner hashing interface, like by solving domain seperation.  It's [STROBE](https://strobe.sourceforge.io/) under the hood, which uses Keccak.